File: proxy-server.js
// Proxy simple para FiveM: evita CORS reenviando a IP:PUERTO
// Uso: node proxy-server.js --target http://IP:PUERTO --port 5050

const http = require('http');
const https = require('https');
const { URL } = require('url');

const args = process.argv.slice(2);
function arg(name, def){
  const i = args.indexOf(`--${name}`);
  if(i === -1) return def;
  const v = args[i+1];
  return v ?? def;
}

const target = new URL(arg('target', 'http://127.0.0.1:30120'));
const port = Number(arg('port', '5050'));
const prefix = arg('prefix', '/fivem');

const agent = target.protocol === 'https:' ? https : http;

const server = http.createServer((req, res)=>{
  // Solo permitimos los endpoints FiveM conocidos
  const allowed = ['/dynamic.json', '/info.json', '/players.json'];
  const urlPath = req.url.replace(prefix, '') || '/';
  if(!allowed.includes(urlPath)){
    res.writeHead(404, {'Content-Type':'application/json', 'Access-Control-Allow-Origin':'*'});
    res.end(JSON.stringify({error:'Not found'}));
    return;
  }
  const proxUrl = new URL(urlPath, target);
  const opts = { method: 'GET', headers: { 'User-Agent': 'RealityRP-Proxy' } };
  const p = agent.request(proxUrl, opts, (pr)=>{
    const headers = { ...pr.headers, 'access-control-allow-origin': '*' };
    res.writeHead(pr.statusCode || 500, headers);
    pr.pipe(res);
  });
  p.on('error', (err)=>{
    res.writeHead(502, {'Content-Type':'application/json', 'Access-Control-Allow-Origin':'*'});
    res.end(JSON.stringify({error:'Bad gateway', detail: String(err)}));
  });
  p.end();
});

server.listen(port, ()=>{
  console.log(`[Proxy] Escuchando en http://localhost:${port}${prefix} -> ${target.href}`);
});

File: proxy_server.py
#!/usr/bin/env python3
import http.server
import socketserver
import urllib.request
import urllib.error
import sys
from urllib.parse import urljoin

PORT = 5050
TARGET = 'http://79.116.6.42:30120'
PREFIX = '/fivem'
ALLOWED = {'/dynamic.json', '/info.json', '/players.json'}

class ProxyHandler(http.server.BaseHTTPRequestHandler):
    def do_GET(self):
        path = self.path
        if not path.startswith(PREFIX):
            self.send_response(404)
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(b'Not found')
            return
        upstream_path = path[len(PREFIX):] or '/'
        if upstream_path not in ALLOWED:
            self.send_response(404)
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(b'Not found')
            return
        url = urljoin(TARGET, upstream_path)
        try:
            req = urllib.request.Request(url, headers={'User-Agent':'RealityRP-PyProxy'})
            with urllib.request.urlopen(req, timeout=8) as resp:
                data = resp.read()
                self.send_response(resp.status)
                for k, v in resp.getheaders():
                    # evitar header hop-by-hop y asegurar CORS
                    if k.lower() in {'transfer-encoding', 'connection', 'keep-alive', 'proxy-authenticate', 'proxy-authorization', 'te', 'trailer', 'upgrade'}:
                        continue
                    self.send_header(k, v)
                self.send_header('Access-Control-Allow-Origin', '*')
                self.end_headers()
                self.wfile.write(data)
        except urllib.error.HTTPError as e:
            self.send_response(e.code)
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(e.read())
        except Exception as e:
            self.send_response(502)
            self.send_header('Content-Type', 'application/json')
            self.send_header('Access-Control-Allow-Origin', '*')
            self.end_headers()
            self.wfile.write(('{"error":"Bad gateway","detail":"%s"}' % str(e)).encode('utf-8'))

def run(port):
    with socketserver.TCPServer(('', port), ProxyHandler) as httpd:
        sa = httpd.socket.getsockname()
        print(f"[PyProxy] Escuchando en http://localhost:{sa[1]}{PREFIX} -> {TARGET}")
        httpd.serve_forever()

if __name__ == '__main__':
    # Permitir override por CLI: python proxy_server.py 5050 http://IP:PORT
    if len(sys.argv) >= 2:
        try:
            PORT = int(sys.argv[1])
        except: pass
    if len(sys.argv) >= 3:
        TARGET = sys.argv[2]
    run(PORT)
